google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/snippets/PublisherClientSnippets.java

```diff
@@ -0,0 +1,210 @@
+      for (Topic topic : topics) {
+        // do something with the topic
+      }
+      // [END listTopics]
+      return response;
+    }
+  }
+
+  /** Example of listing topics, specifying the page size and a page token. */
+  public ListTopicsPagedResponse listTopicsWithPageToken(String pageToken) throws Exception {
+    try (PublisherClient publisherClient = PublisherClient.create()) {
+      // [START listTopicsWithPageToken]
+      ListTopicsRequest listTopicsRequest =
+          ListTopicsRequest.newBuilder()
+              .setProjectWithProjectName(ProjectName.create(projectId))
+              .setPageSize(100)
+              .setPageToken(pageToken)
+              .build();
+      ListTopicsPagedResponse response =
+          publisherClient.listTopics(listTopicsRequest);
+      Iterable<Topic> topics = response.iterateAllElements();
+      for (Topic topic : topics) {
+        // do something with the topic
+      }
+      // [END listTopicsWithPageToken]
+      return response;
+    }
+  }
+
+  /** Example of listing topics for a subscription, specifying the page size. */
+  public ListTopicSubscriptionsPagedResponse listTopicSubscriptions(String topicName) throws Exception {
+    try (PublisherClient publisherClient = PublisherClient.create()) {
+      TopicName formattedTopicName = TopicName.create(projectId, topicName);
+      // [START listTopicSubscriptions]
+      ListTopicSubscriptionsRequest request =
+          ListTopicSubscriptionsRequest.newBuilder()
+              .setTopicWithTopicName(formattedTopicName)
+              .setPageSize(100)
+              .build();
+      ListTopicSubscriptionsPagedResponse response =
+          publisherClient.listTopicSubscriptions(request);
+      Iterable<String> subscriptions = response.iterateAllElements();
+      for (String subscription : subscriptions) {
+        // do something with the subscription name
+      }
+      // [END listTopicSubscriptions]
+      return response;
+    }
+  }
+
+  /** Example of listing topics for a subscription, specifying the page size and page token */
+  public ListTopicSubscriptionsPagedResponse listTopicSubscriptionsWithPageToken(
+      String topicName, String pageToken) throws Exception {
+    try (PublisherClient publisherClient = PublisherClient.create()) {
+      TopicName formattedTopicName = TopicName.create(projectId, topicName);
+      // [START listTopicSubscriptionsWithPageToken]
+      ListTopicSubscriptionsRequest request =
+          ListTopicSubscriptionsRequest.newBuilder()
+              .setTopicWithTopicName(formattedTopicName)
+              .setPageSize(100)
+              .setPageToken(pageToken)
+              .build();
+      ListTopicSubscriptionsPagedResponse response =
+          publisherClient.listTopicSubscriptions(request);
+      Iterable<String> subscriptions = response.iterateAllElements();
+      for (String subscription : subscriptions) {
+        // do something with the subscription name
+      }
```

Comment:
I don't think it is a typical use case to use a page token, and then iterate over the remainder of the pages. Usually you would either iterate over everything (ignore page tokens), or iterate page by page. In fact, it looks like there is no example for page-by-page iteration.